_blog/what-is-kubernetes/index.md

```diff
@@ -0,0 +1,469 @@
+
+Each virtual machine comes with an operating system that consumes part of the memory and CPU resources allocated to it.
+
+When you create a 1GB of memory and 1 vCPU droplet on Digital Ocean, you end up using 700MB in memory and 0.8 vCPU after you remove the overhead of the operating system.
+
+{% include_relative overhead.html %}
+
+Or in other words, every fifth virtual machine the overhead adds up to a full virtual machine.
+
+**You pay for five but can use only four.**
+
+You can't escape from it, even if you're on bare metal.
+
+You still need to run your services from a base operating system.
+
+_It's okay, everybody needs to run an operating system_ — you say.
+
+And you're right.
+
+However, the cash wasted on operating systems is only the tip of the iceberg.
+
+## You're also wasting A LOT of money on resource utilisation
+
+You have probably realised that when you break your service into smaller components, each of them comes with different resource requirements.
+
+Some components such as data processing and data mining applications are CPU intensive.
+
+Others, such as servers for real-time applications might use more memory than CPU.
+
+{% include_relative profiles.html %}
+
+Amazon Web Services and the other cloud providers have indeed a long list of compute resources that fit every need: general purpose, CPU optimised, memory optimised, storage optimised and GPU computing.
+
+You should strive to use the right virtual machine for your component.
+
+Ideally, it should match the memory consumption and CPU utilisation.
+
+_Are you working on a critical web component written in Java?_
+
+Perhaps you should use a c5.4xlarge optimised for compute-intensive workloads.
+
+The closer you match the requirements, the better you're utilising your resources.
+
+In practice, this is somewhat uncommon, though.
+
+_Should you use a c5.2xlarge or a c5.4xlarge?_
+
+_Does the next tier (8 vCPU and 16GB memory) make a difference?_
+
+It's much easier to select a couple of computing profiles that are good enough in 80% of the cases and use those for all components.
+
+_In fact, what's wrong with using mostly the same virtual machine for every workload?_
+
+None at all, if you're happy to wrap every component into a 2GB of memory and vCPU compute capacity.
+
+Even if your component can run with only 1GB of memory.
+
+_Yes, you could optimise in the future._
+
+But let's be honest: **it's like changing tyres while driving**.
+
+You put a lot of effort into tuning the system only to realise that the application changed again and you have to start from scratch.
+
+So you end up taking the only sensible choice: selecting a small, medium and large profile for virtual machines and using them for all workloads.
+
+You know you have to live with wasting hundreds of megabytes of RAM and plenty of CPU cycles.
+
+{% include_relative waste.html %}
+
+If it makes you feel any better, there're plenty of companies suffering from similar inefficiencies.
```

Comment:
I'm not using Grammarly but this sounds off _there're_. I don't think this abbreviation is grammatically correct.

Reply:
I can have `there are`. Is that what you meant?

Reply:
Yes, that's what I mean.